main.c

```diff
@@ -354,8 +354,34 @@ void move_bug(
     struct board_tile board[SIZE][SIZE],
     struct bug_node* bug_linked_list
 ) {
-    printf("// TODO move_bug\n");
-    // Hint: use bug_linked_list to traverse over the list of bug
+    struct bug_node* current_bug = bug_linked_list;
+    while (current_bug != NULL) {
+        int next_col = current_bug->col;
+        int direction = board[current_bug->row][current_bug->col].type == WATER ? 1 : -1; // move right by default
+
+        if (direction == 1 && current_bug->col < RIGHT_COLUMN
+            && board[current_bug->row][current_bug->col + 1].bug == NULL)
```

Comment:
need to check that it's either Turtle or Log as well
```suggestion
            && (board[current_bug->row][current_bug->col + 1].type == TURTLE || board[current_bug->row][current_bug->col + 1].type == LOG)
            && board[current_bug->row][current_bug->col + 1].bug == NULL)
```